packages/fxa-auth-server/test/local/routes/subscriptions.js

```diff
@@ -0,0 +1,409 @@
+              plan_id: PLANS[0].plan_id,
+              token: PAYMENT_TOKEN_VALID,
+            },
+          }
+        );
+        assert.fail();
+      } catch (err) {
+        assert.deepEqual(err.errno, error.ERRNO.BACKEND_SERVICE_FAILURE);
+        assert.equal(db.createAccountSubscription.callCount, 0);
+      }
+    });
+
+    it('should correctly handle an unknown plan', async () => {
+      try {
+        await runTest(
+          '/subscriptions/active',
+          {
+            ...requestOptions,
+            method: 'POST',
+            payload: {
+              ...requestOptions.payload,
+              plan_id: 'thisisnotaplan',
+              token: PAYMENT_TOKEN_VALID,
+            },
+          }
+        );
+        assert.fail();
+      } catch (err) {
+        assert.deepEqual(err.errno, error.ERRNO.UNKNOWN_SUBSCRIPTION_PLAN);
+        assert.equal(db.createAccountSubscription.callCount, 0);
+      }
+    });
+
+    it('should correctly handle payment token rejection', async () => {
+      subhub.createSubscription =
+        sinon.spy(async (uid, token, plan_id) => null);
+      try {
+        await runTest(
+          '/subscriptions/active',
+          {
+            ...requestOptions,
+            method: 'POST',
+            payload: {
+              ...requestOptions.payload,
+              plan_id: PLANS[0].plan_id,
+              token: PAYMENT_TOKEN_VALID,
+            },
+          }
+        );
+        assert.fail();
+      } catch (err) {
+        assert.equal(err.errno, error.ERRNO.REJECTED_SUBSCRIPTION_PAYMENT_TOKEN);
+        assert.equal(db.createAccountSubscription.callCount, 0);
+      }
+    });
+  });
+
+  describe('POST /subscriptions/updatePayment', () => {
+    it('should allow updating of payment method', async () => {
+      // TODO: TBD subhub response for updatePayment, do something with it?
+      await runTest(
+        '/subscriptions/updatePayment',
+        {
+          ...requestOptions,
+          method: 'POST',
+          payload: { token: PAYMENT_TOKEN_NEW }
+        }
+      );
+      assert.equal(customs.check.callCount, 1, 'calls customs.check');
+      assert.deepEqual(
+        subhub.updateCustomer.args,
+        [ [ UID, PAYMENT_TOKEN_NEW ] ]
+      );
+    });
+
+    it('should correctly handle subscription backend failure', async () => {
+      subhub.updateCustomer = sinon.spy(async () => {
+        throw error.backendServiceFailure();
+      });
+      try {
+        await runTest(
+          '/subscriptions/updatePayment',
+          {
+            ...requestOptions,
+            method: 'POST',
+            payload: { token: PAYMENT_TOKEN_NEW }
+          }
+        );
+        assert.fail();
+      } catch (err) {
+        assert.deepEqual(err.errno, error.ERRNO.BACKEND_SERVICE_FAILURE);
+        assert.equal(db.createAccountSubscription.callCount, 0);
+      }
+    });
+
+    it('should correctly handle payment token rejection', async () => {
```

Comment:
You can also testing for updating an unknown subscription.

Reply:
This method's just for creating a subscription, so we're really just dealing with a plan & payment here so far.